### PR TITLE
HPCC-20815 Add expireDays attribute to Logical Files

### PR DESCRIPTION
**Pull request by kunalaswani**
Add the expireDays attribute to the Logical Files.

Signed-off-by: Aswani, Kunal <kunal.aswani@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [x] I have tested my changes in multiple modern browsers
  - [x] The component(s) render as expected

### CODE REVIEW

**Comment by kunalaswani** [comment]
@miguelvazq please review

**Comment by richardkchapman** [comment]
@kunalaswani I changed the target to 7.2.0 - can you run

rebase --onto candidate-7.2.0 master

then force-push

**Comment by kunalaswani** [comment]
@richardkchapman rebased @miguelvazq ready for review, tested and looks to be working as expected.

File: esp/src/eclwatch/LFDetailsWidget.js [hunk 1]
```diff
@@ -385,6 +385,8 @@ define([
                     dom.byId(this.id + "ProtectedImage").src = this.logicalFile.getProtectedImage();
                 } else if (name === "IsCompressed") {
                     dom.byId(this.id + "CompressedImage").src = this.logicalFile.getCompressedImage();
+                } else if (name === "ExpireDays") {
+                    dom.byId(this.id + "ExpireDays").src = this.logicalFile.ExpireDays();
```
[the remark this answers is not in the document]
Comment (miguelvazq) [reply esp/src/eclwatch/LFDetailsWidget.js]: There is no need to add an else if here unless you wanted to fetch the value from somewhere else based on an event or some logic. The `switch` statement above should automatically add the value to your HTML element.

**Review by miguelvazq — APPROVED** [review]

**Comment by kunalaswani** [comment]
@AttilaVamos can you check if this failure is due to roxie, or due to my code?

**Comment by AttilaVamos** [comment]
@kunalaswani It is not related to this PR.

**Pull request merged**